e2e/README.md

```diff
@@ -78,4 +78,64 @@ These logs will be uploaded in a bundle of the format:
         ├── vmssId.txt
 ```
 
+## Coverage report
+
+After a PR is created in Agent Baker's repo on GitHub, pipeline calculating code coverage changes will automatically run.
+
+We are utilizing [coveralls](https://coveralls.io/) to view the coverage report. The coverage report will be available in the PR's description.
+You can also view previous runs for AgentBaker repo [here](https://coveralls.io/github/Azure/AgentBaker). 
+
+We calculate code coverage for both unit tests and E2E tests.
+
+### E2E coverage report
+
+To calculate E2E coverage report we are using code [coverage changes](https://go.dev/blog/integration-test-coverage) introduced in Go 1.20.
+
+Coverage report is generated by running Agent Baker's API server locally as a binary created with -cover flag.
+E2E tests are then ran against that binary.
+
+Following packages are used during calculation of coverage for E2E tests:
+````
+- github.com/Azure/agentbaker/apiserver  
+- github.com/Azure/agentbaker/cmd 
+- github.com/Azure/agentbaker/cmd/starter
+- github.com/Azure/agentbaker/pkg/agent
+- github.com/Azure/agentbaker/pkg/agent/datamodel
+- github.com/Azure/agentbaker/pkg/templates
+````
+
+#### Generating E2E coverage report locally
+You can generate E2E coverage report while running the E2E tests locally. 
+To do so, follow the steps below:
+
+1. Build Agent Baker server binary with -cover flag:
+```bash
+  cd cmd 
+  go build -cover -o baker -covermode count 
+  GOCOVERDIR=covdatafiles ./baker start &
+```
+2. Create directory for coverage report files
+```bash
+  mkdir -p covdatafiles
+```     
+3. Run the binary
+```bash
+  GOCOVERDIR=covdatafiles ./baker start &
+```
+4. Run the E2E tests locally
+```bash
+  /bin/bash e2e/e2e-local.sh
+```
+5. Stop the binary - once the test finish executing, you have to stop the binary with exit code 0 to generate the report.
```

Comment:
do you mean here that the binary needs to exit with exit code 0 for the report to be generated?